utils/hdparm.go

```diff
@@ -207,6 +211,136 @@ func (h *Hdparm) DriveCapabilities(ctx context.Context, logicalName string) ([]*
 	return capabilities, err
 }
 
+// WipeDrive implements DriveWipe by calling Sanitize or Erase as appropriate.
+// Sanitize(CryptoErase) is preferred over Sanitize(BlockErase) which is preferred over Erase(CryptographicErase).
+func (h *Hdparm) WipeDrive(ctx context.Context, logger *logrus.Logger, drive *common.Drive) error { // nolint:gocyclo
+	var (
+		esee     bool
+		eseu     bool
+		sanitize bool
+		bee      bool
+		cse      bool
+	)
+	for _, cap := range drive.Capabilities {
+		switch {
+		case cap.Description == "encryption supports enhanced erase":
+			esee = cap.Enabled
+		case cap.Description == "BLOCK ERASE EXT":
+			bee = cap.Enabled
+		case cap.Description == "CRYPTO SCRAMBLE EXT":
+			cse = cap.Enabled
+		case cap.Description == "SANITIZE feature":
+			sanitize = cap.Enabled
+		case strings.HasPrefix(cap.Description, "erase time:"):
+			eseu = strings.Contains(cap.Description, "enhanced")
+		}
+	}
+
+	l := logger.WithField("drive", drive.LogicalName)
+	if sanitize && cse {
+		// nolint:govet
+		l := l.WithField("method", "sanitize").WithField("action", "sanitize-crypto-scramble")
+		l.Info("wiping")
```

Comment:
Can this log at debug level instead, it would be ideal to avoid info level logging in the library

Reply:
This is useful info that I do not want to hide behind both debug level and potentially lots of log messages. With both FillZeros usually taking a long time and both hdparm and nvme wipes introspecting the capabilities and potentially trying multiple wipe methods/actions I think logging these makes sense. If the user does not want to see them then they can pass in a logger that will filter out info messages. I guess the same could be said for doing at debug too :D. Ok you've convinced me :), but I'd like to do that in a follow up PR since the other WiperDrive functions are currently logging at info level and I'd like to do them all in one PR.

Reply:
Theres other code in ironlib thats logging at info level as well (that would be my fault as well).
Logging is not often the behaviour we want from a library, but I can see with destructive/long running actions like wiping drives this can be necessary. My suggestion was more of a guideline, feel free to change things around when it becomes noisy.

Reply:
Agree about logging, I have the same feeling. FillZero's progress output was the push for logging in the wipe method and I just went with info since I was being explicit about passing in the logger instead of embedding/re-using previous one I figured it wasn't that bad of an idea, but yeah I'll follow up with a PR for changing this.